projects/common-platform-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/client/OsClient.kt

```diff
@@ -0,0 +1,130 @@
+package uk.gov.justice.digital.hmpps.integrations.client
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.service.annotation.GetExchange
+
+interface OsClient {
+    @GetExchange("/search/places/v1/find")
+    fun searchByFreeText(
+        @RequestParam query: String,
+        @RequestParam maxResults: Int,
+        @RequestParam key: String
+    ): OsPlacesResponse
+}
+
+data class OsPlacesResponse(
+    val header: OsPlacesHeader,
+    val results: List<DpaWrapper>?
+)
+
+data class OsPlacesHeader(
+    val uri: String?,
+    val query: String?,
+    val offset: Int?,
+    val totalresults: Int?,
+    val format: String?,
+    val dataset: String?,
+    val lr: String?,
+    val maxresults: Int?,
+    val matchprecision: Double?,
+    val filter: String?,
+    val srs: String?,
+    val epoch: String?,
+    val lastupdate: String?,
+    val output_srs: String?
+)
+
+class DpaWrapper {
+    @JsonProperty("DPA")
+    val dpa: Dpa? = null
+}
+
+data class Dpa(
+    @JsonProperty("UPRN")
+    val uprn: String?,
```

Comment:
Are all these fields nullable?  Seems like some of these have a non-zero "multiplicity" in the docs - which I guess means they'll always be returned? e.g. POST_TOWN: https://osdatahub.os.uk/docs/places/technicalSpecification#:~:text=0..1-,POST_TOWN,-The%20town%20or